modules/ROOT/pages/java-support.adoc

```diff
@@ -169,6 +169,27 @@ If you are a MuleSoft customer:
 . Test your connector using xref:customer-connector-upgrade.adoc#test-your-custom-connector-with-munit[MUnit].
 . xref:mule-runtime::deploying.adoc[Deploy your applications] to a target running on Mule 4.6 or later.
 
+[[custom-code-in-connectors]]
+== Custom Code in Connectors 
+
+If you have custom code in your connectors, or if you are using third-party libraries, you must update that custom code to be compatible with Java 17.
+
+* Reflection on Java classes
++
+To ensure compatibility with Java 17, review and update any code that uses reflection to access Java classes or set parameter values. Java 17 imposes stricter access controls, so you should avoid introspecting JDK internals. Ensure you have a robust test suite and make sure it is run with Java 17. Pay special attention to the `setAccessible(true)` method because these type of changes in the accessibility is not allowed in Java 17.
+
+* Java Platform Module System (JPMS)
++
+JPMS was introduced in Java 9. This feature introduces stricter encapsulation, impacting how Mule modules interact with JDK classes. As Mule upgrades to Java 17 and JPMS, the following changes are necessary:
++
+** *Refactor Split Packages*: Resolve issues with internal and API package splits by reorganizing and refactoring packages to conform with JPMS modularization standards.
+** *Address ResourceBundle Loading*: Implement solutions for issues related to resource bundle loading, such as using `ResourceBundleProvider` or alternative methods.
+** Analyze the third-party libraries' changelogs to understand the changes related to JPMS or about Java 17 compatibility.
```

Comment:
I'm reading this part 2-3 times: "changes related to JPMS or about Java 17 compatibility" 
though I understand the reason for adding JPMS first, the "about Java 17..." part impacts the flow when I read it

```suggestion
** Analyze the third-party libraries' changelogs to understand Java 17 compatibility and changes related to JPMS.
```